documentation/design-docs/ipc-protocol.md

```diff
@@ -473,36 +496,134 @@ Payload
 }
 ```
 
-### Errors
+## Dump Commands
 
-In the event an error occurs in the handling of an Ipc Message, the Diagnostic Server will attempt to send an Ipc Message encoding the error and subsequently close the connection.  The connection will be closed **regardless** of the success of sending the error message.  The Client is expected to be resilient in the event of a connection being abruptly closed.
+### `CreateCoreDump`
 
-```c++
-enum class DiagnosticServerErrorCode : uint32_t
+Command Code: `0x0101`
+
+The `CreateCoreDump` command is used to instruct the runtime to generate a core dump of the process.  The command will keep the connection open while the dump is generated and then response with a message containing an `HRESULT` indicating success or failure.
+
+In the event of an [error](#Errors), the runtime will attempt to send an error message and subsequently close the connection.
+
+#### Inputs:
+
+Header: `{ Magic; Size; 0x0101; 0x0000 }`
+
+* `string dumpName`: The name of the dump generated.
+* `uint dumpType`: A value between 1 and 4 inclusive that indicates the type of dump to take
+  * Normal = 1,
+  * WithHeap = 2,
+  * Triage = 3,
+  * Full = 4
+* `uint diagnostics`: The providers to turn on for the streaming session
+  * `0` or `1` for on or off
+
+#### Returns (as an IPC Message Payload):
+
+Header: `{ Magic; 28; 0xFF00; 0x0000; }`
+
+`CreateCoreDump` returns:
+* `int32 hresult`: The result of creating the core dump (`0` indicates success)
+
+##### Details:
+
+Input:
+```
+Payload
 {
-    OK                = 0x00000000,
-    BadEncoding       = 0x00000001,
-    UnknownCommand    = 0x00000002,
-    UnknownMagic      = 0x00000003,
-    BadInput          = 0x00000004,
-    // future
+    string dumpName,
+    uint dumpType,
+    uint diagnostics
+}
+```
 
-    UnknownError      = 0xFFFFFFFF,
-};
+Returns:
+```c
+Payload
+{
+    int32 hresult
+}
+```
+
+## Profiler Commands
+
+### `AttachProfiler`
+
+Command Code: `0x0301`
+
+The `AttachProfiler` command is used to attach a profiler to the runtime.  The command will keep the connection open while the profiler is being attached and then response with a message containing an `HRESULT` indicating success or failure.
+
+In the event of an [error](#Errors), the runtime will attempt to send an error message and subsequently close the connection.
+
+#### Inputs:
+
+Header: `{ Magic; Size; 0x0301; 0x0000 }`
+
+* `uint32 attachTimeout`: The timeout for attaching to the profiler (in milliseconds)
+* `CLSID profilerGuid`: The GUID associated with the profiler
+* `string profilerPath`: Location of the profiler
+* `array<byte> clientData`: The data being provided to the profiler
+
+Where a `CLSID` is a fixed size struct consisting of:
+* `uint x`
+* `byte s1`
+* `byte s2`
+* `byte[8] c`
+
+#### Returns (as an IPC Message Payload):
+
+Header: `{ Magic; 28; 0xFF00; 0x0000; }`
+
+`AttachProfiler` returns:
+* `int32 hresult`: The result of creating the core dump (`0` indicates success)
```

Comment:
You mean result of attaching the profiler?

Reply:
```suggestion
* `int32 hresult`: The result of attaching the profiler (`0` indicates success)
```